Tasks/AndroidSigningV3/androidsigning.ts

```diff
@@ -1,19 +1,39 @@
 import * as path from 'path';
 import * as tl from 'azure-pipelines-task-lib/task';
+import * as semver from 'semver';
 
 const findAndroidTool = (tool: string): string => {
     const androidHome = tl.getVariable('ANDROID_HOME');
     if (!androidHome) {
         throw new Error(tl.loc('AndroidHomeNotSet'));
     }
 
-    // add * in search as on Windows the tool may end with ".exe" or ".bat"
-    const toolsList = tl.findMatch(tl.resolve(androidHome, 'build-tools'), tool + '*', null, { matchBase: true });
+    // add * in search as on Windows the tool may end with ".exe" or ".bat. Exclude .jar files from the list"
+    const toolsList = tl.findMatch(tl.resolve(androidHome, 'build-tools'), [`${tool}*`, "!*.jar"], null, { matchBase: true })
 
     if (!toolsList || toolsList.length === 0) {
         throw new Error(tl.loc('CouldNotFindToolInAndroidHome', tool, androidHome));
     }
 
+    // use the latest version of build-tools?
+    const useLatestBuildTools: boolean = tl.getBoolInput('useLatestBuildTools');
+
+    // if useLatestBuildTools is set, sort toolsList descending
+    if(useLatestBuildTools) {
```

Comment:
Can we cover this logic by unit-tests? Also, it will be good to add a new canary test to cover a new usage scenario.

Reply:
Thank you! Covered with unit tests.